test_key_def_structure gives FileNotFoundError

**Describe the bug**
`test_key_def_structure` failed on main branch (https://github.com/equinor/ert/runs/5676295165?check_suite_focus=true), but only for Python 3.8 on macos. Error:

```python
____________________________ test_key_def_structure ____________________________

api = <ert_gui.tools.plot.plot_api.PlotApi object at 0x149ece4c0>

    def test_key_def_structure(api):
>       shutil.rmtree("storage")

/Users/runner/work/ert/ert/tests/ert_tests/gui/tools/plot/test_plot_api.py:222: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/runner/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/shutil.py:718: in rmtree
    _rmtree_safe_fd(fd, path, onerror)
/Users/runner/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/shutil.py:655: in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
/Users/runner/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/shutil.py:655: in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
/Users/runner/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/shutil.py:655: in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
/Users/runner/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/shutil.py:675: in _rmtree_safe_fd
    onerror(os.unlink, fullname, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

topfd = 244, path = 'storage/snake_oil/ensemble/default_0'
onerror = <function rmtree.<locals>.onerror at 0x14ac1e5e0>

    def _rmtree_safe_fd(topfd, path, onerror):
        try:
            with os.scandir(topfd) as scandir_it:
                entries = list(scandir_it)
        except OSError as err:
            err.filename = path
            onerror(os.scandir, path, sys.exc_info())
            return
        for entry in entries:
            fullname = os.path.join(path, entry.name)
            try:
                is_dir = entry.is_dir(follow_symlinks=False)
            except OSError:
                is_dir = False
            else:
                if is_dir:
                    try:
                        orig_st = entry.stat(follow_symlinks=False)
                        is_dir = stat.S_ISDIR(orig_st.st_mode)
                    except OSError:
                        onerror(os.lstat, fullname, sys.exc_info())
                        continue
            if is_dir:
                try:
                    dirfd = os.open(entry.name, os.O_RDONLY, dir_fd=topfd)
                except OSError:
                    onerror(os.open, fullname, sys.exc_info())
                else:
                    try:
                        if os.path.samestat(orig_st, os.fstat(dirfd)):
                            _rmtree_safe_fd(dirfd, fullname, onerror)
                            try:
                                os.rmdir(entry.name, dir_fd=topfd)
                            except OSError:
                                onerror(os.rmdir, fullname, sys.exc_info())
                        else:
                            try:
                                # This can only happen if someone replaces
                                # a directory with a symlink after the call to
                                # os.scandir or stat.S_ISDIR above.
                                raise OSError("Cannot call rmtree on a symbolic "
                                              "link")
                            except OSError:
                                onerror(os.path.islink, fullname, sys.exc_info())
                    finally:
                        os.close(dirfd)
            else:
                try:
>                   os.unlink(entry.name, dir_fd=topfd)
E                   FileNotFoundError: [Errno 2] No such file or directory: 'default_0.lock'

/Users/runner/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/shutil.py:673: FileNotFoundError
```

**To Reproduce**
TBD

**Expected behavior**
Not fail

**Enviromment**
 - OS: Mac
 - ERT/Komodo Release: main (https://github.com/equinor/ert/commit/dae619a3a9827f15a6d559a1aecebb12f422ec87)
 - Python version: 3.8
 - Remote/HPC execution involved: no

--- a/tests/ert_tests/gui/tools/plot/test_plot_api.py
+++ b/tests/ert_tests/gui/tools/plot/test_plot_api.py
@@ -219,7 +219,7 @@ class PlotApiTest(TestCase):
 
 
 def test_key_def_structure(api):
-    shutil.rmtree("storage")
+    shutil.rmtree("storage", ignore_errors=True)
     key_defs = api.all_data_type_keys()
     fopr = next(x for x in key_defs if x["key"] == "FOPR")
 

